Jint/Native/Array/ArrayPrototype.cs

```diff
@@ -170,14 +165,16 @@ private JsValue Filter(JsValue thisObj, JsValue[] arguments)
             var a = (ArrayInstance)Engine.Array.Construct(Arguments.Empty);
 
             var to = 0;
+            var jsValues = new JsValue[3];
```

Comment:
Cache that as local thread stuff as well?
For that matter, might as well create some shared cache per thread for all of that and have a property such as `List`, `3 element array`, etc.

Reply:
Before I do own benchmarks, would you happen to know performance benefits between small array allocation and thread local data access? What I've understood small arrays are quite efficient, but probably worth checking out.